clientcontroller/evm_consumer.go

```diff
@@ -81,34 +86,140 @@ func (ec *EVMConsumerController) QueryFinalityProviderVotingPower(fpPk *btcec.Pu
 }
 
 func (ec *EVMConsumerController) QueryLatestFinalizedBlocks(count uint64) ([]*types.BlockInfo, error) {
-	return ec.queryLatestBlocks(nil, count, finalitytypes.QueriedBlockStatus_FINALIZED, true)
+
+	lastNumber, err := ec.GetLatestFinalizedNumber()
+	if err != nil {
+		return nil, fmt.Errorf("can't get latest finalized block:%s", err)
+
+	}
+
+	type Block struct {
+		Number string
+		Hash   string
+	}
+
+	var batchArray []rpc.BatchElem
+
+	initBatch(batchArray, lastNumber, count, "descent")
+
+	err = ec.evmClient.BatchCall(batchArray)
+
+	if err != nil {
+		return nil, fmt.Errorf("can't get latest block:%s", err)
+
+	}
+	var blocks []*types.BlockInfo
+
+	for _, batch := range batchArray {
+		nb := batch.Result.(*Block)
+		num, err := strconv.ParseUint(strings.TrimPrefix(nb.Number, "0x"), 16, 64)
+		if err != nil {
+			fmt.Println("Error:", err)
+		}
+		block := &types.BlockInfo{
+			Height: num,
+			Hash:   []byte(nb.Hash),
+		}
+		blocks = append(blocks, block)
+	}
+	return blocks, nil
 }
 
 func (ec *EVMConsumerController) QueryBlocks(startHeight, endHeight, limit uint64) ([]*types.BlockInfo, error) {
 
-	return ec.queryLatestBlocks(sdk.Uint64ToBigEndian(startHeight), 0, finalitytypes.QueriedBlockStatus_ANY, false)
+	if endHeight < startHeight {
+		return nil, fmt.Errorf("the startHeight %v should not be higher than the endHeight %v", startHeight, endHeight)
+	}
+	count := endHeight - startHeight
+	if count > limit {
+		count = limit
+	}
+
+	type Block struct {
+		Number string
+		Hash   string
+	}
+
+	startnumber := new(big.Int).SetUint64(startHeight)
+
+	var batchArray []rpc.BatchElem
+
+	initBatch(batchArray, startnumber, count, "ascent")
+
+	err := ec.evmClient.BatchCall(batchArray)
+
+	if err != nil {
+		return nil, fmt.Errorf("can't get blocks")
+
+	}
+
+	var blocks []*types.BlockInfo
+
+	for _, batch := range batchArray {
+		nb := batch.Result.(*Block)
+		num, err := strconv.ParseUint(strings.TrimPrefix(nb.Number, "0x"), 16, 64)
```

Comment:
this looks very wrong. query a block should be very simply and standard operation

the entire implementation of this function should be concise